06-data-subsetting.md

```diff
@@ -1318,3 +1331,29 @@ be changed with the third argument, `drop = FALSE`).
 > gapminder_small <- gapminder[c(1:9, 19:23),]
 > ~~~
 >
+
+> ## Solution to challenge 7 {.challenge}
+>
+> How many countries in Asia had a population in either 2002 or 2007 above 20 million? 
+> Tip: Use subsetting to maintain the relevant data and then use the function `unique()` to remove duplicate countries.
+>
+> A. 19
+>
+> B. 20
+>
+> C. 21
+>
+> D. 22
+>
+> Answer: B
+> 
+> The answer can be derived using the following statement
+> ~~~{.r}
+> length(unique(
+>   gapminder[
+>     gapminder$year >= 2002 & 
+>     gapminder$year <= 2007 & 
```

Comment:
This may be as good a place as any to talk about the differences between `&` vs `&&` and `|` vs `||` in R. From the help docs:

> & and && indicate logical AND and | and || indicate logical OR. The shorter form performs elementwise comparisons in much the same way as arithmetic operators. The longer form evaluates left to right examining only the first element of each vector. Evaluation proceeds only until the result is determined. The longer form is appropriate for programming control-flow and typically preferred in if clauses.

This is often something people (especially those coming from other languages) want to know and get tripped up on.